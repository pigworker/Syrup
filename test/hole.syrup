not(<Bit>) -> <Bit>
not(X) = nand(X,X)

and(<Bit>, <Bit>) -> <Bit>
and(X,Y) = !nand(X,Y)

or(<Bit>, <Bit>) -> <Bit>
or(X,Y) = nand(!X,!Y)


f(<Bit>) -> <Bit>
f(X) = !R & S & ?A & (?B | ?C) where

  [R,S] = ?D

problematic() -> <Bit>
problematic() = ?