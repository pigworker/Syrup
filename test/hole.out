Circuit not is defined.

Circuit and is defined.

Circuit or is defined.

Found holes in circuit f:
  ?A : <Bit>
  ?B : <Bit>
  ?C : <Bit>
  ?D : [<Bit>, <Bit>]

Warning: Circuit f has been stubbed out.

Error: I was trying to make sense of the following code:

problematic() = ?

I got stuck here: 
  problematic() =  {HERE} ? 
I was looking for a definition.
At that point, it made sense, but then I found stuff I didn't expect.

Error: You haven't defined the circuit problematic just now.

Warning: Circuit problematic has been stubbed out.

